src/components/pill-selector-dropdown/PillSelector.js

```diff
@@ -199,17 +210,39 @@ class PillSelector extends React.Component<Props, State> {
                     onFocus={this.handleFocus}
                     onKeyDown={this.handleKeyDown}
                 >
-                    {selectedOptions.map((option: Option, index: number) => (
-                        <Pill
-                            isValid={allowInvalidPills ? validator(option) : true}
-                            isDisabled={disabled}
-                            isSelected={index === selectedIndex}
-                            key={option.value}
-                            onRemove={onRemove.bind(this, option, index)}
-                            // $FlowFixMe option.text is for backwards compatibility
-                            text={option.displayText || option.text}
-                        />
-                    ))}
+                    {showAvatars
+                        ? selectedOptions.map((option: RoundOption, index: number) => {
+                              return (
+                                  <PillComponent
+                                      isValid={allowInvalidPills ? validator(option) : true}
+                                      isDisabled={disabled}
+                                      isSelected={index === selectedIndex}
+                                      key={option.value}
+                                      onRemove={onRemove.bind(this, option, index)}
+                                      // $FlowFixMe option.text is for backwards compatibility
+                                      text={option.displayText || option.text}
+                                      showAvatar
+                                      id={option.id}
+                                      hasWarning={option.hasWarning}
+                                      isExternal={option.isExternalUser}
+                                  />
+                              );
+                          })
+                        : selectedOptions.map((option: Option, index: number) => {
+                              // TODO: This and associated types will be removed once all views are updates with round pills.
+                              return (
+                                  <PillComponent
+                                      isValid={allowInvalidPills ? validator(option) : true}
+                                      isDisabled={disabled}
+                                      isSelected={index === selectedIndex}
+                                      key={option.value}
+                                      onRemove={onRemove.bind(this, option, index)}
+                                      // $FlowFixMe option.text is for backwards compatibility
+                                      text={option.displayText || option.text}
+                                  />
+                              );
+                          })}
```

Comment:
Instead, could we spread an object of properties if `showAvatar` is true? Something like:
```
const avatarProps = {
  id: option.id,
  hasWarning: option.hasWarning,
  isExternal: option.isExternalUser,
};

const pill = <PillComponent ... />;

return showAvatars ? React.cloneElement(pill, avatarProps) : pill;
```

Reply:
or conditionally construct the props before spreading them on the component?

Reply:
Yea we couldn't group it together because of the different flow type options for when its a RoundedPill vs a Pill. wanted to type the content used in each .map and if you put the map outside, all paths have to use a type without conflicts. suggestions?

Reply:
The desired logic is such that if `showRoundPills` is false, we should always use `Pill. if `showRoundPills` is true, we would or would not use an avatar based on `showAvatars` being passed to `RoundPill`. Given that, we could remove the line for abstracting the pill component name, `.map` based on which pill we want to show, and add `showAvatar={showAvatars}` on ln223 or the equivalent, no?